File: 프로그래머스/0/120843. 공 던지기/공 던지기.java
class Solution {
    public int solution(int[] numbers, int k) {
            int answer = 0;
            int i = 0;
            int count = 1;        
            while (true) {
                if(count > k) break;                
                answer = numbers[i];                    
                if(i == numbers.length - 1 && numbers.length % 2 != 0){
                    i = 0;
                    i += 1;
                }else if(i > numbers.length - 1 && numbers.length % 2 != 0){
                    i = 0;
                    i += 2;
                }else{
                    i += 2;
                    if(i > numbers.length -1) i = 0;
                }                                   
                count++;
            }
            return answer;
        }
}

File: 프로그래머스/0/120843. 공 던지기/공 던지기.py
def solution(numbers, k):
    answer = 0
    i = 0
    count = 1
    
    while True:
        if count > k:
            break
        
        answer = numbers[i]
        
        if i == len(numbers) - 1 and len(numbers) % 2 != 0:
            i = 0
            i += 1
        elif i > len(numbers) - 1 and len(numbers) % 2 != 0:
            i = 0
            i += 2
        else:
            i += 2
            if i > len(numbers) - 1:
                i = 0
        
        count += 1
    
    return answer
